base/compiler/tfuncs.jl

```diff
@@ -1140,15 +1140,15 @@ function fieldtype_tfunc(@nospecialize(s0), @nospecialize(name))
 
     s, exact = instanceof_tfunc(s0)
     s === Bottom && return Bottom
-    return _fieldtype_tfunc(s, exact, name)
+    return _fieldtype_tfunc(s, exact, isa(s0, Const) || hasuniquerep(s0), name)
```

Comment:
The 3 calls to this seem to pass something different to the `s_uniquerep` parameter, which makes me suspicious. Why are some passing `o` and others passing `Type{o}`/`Const(o)` there?